### PR TITLE
added debugDisableShadows usage explaination 

### LINKED ISSUE #69925: debugDisableShadows docs should explain usage
The `debugDisableShadows` property can be used within tests to enable shadows for a given test for screenshots.

The only way this property works is if you set it to `false` in a single test case, then set it back to `true` before the end of the test case. If you try to alter it outside of an individual test, an exception is thrown. If you don't remember to set it back at the end of a test, an exception is thrown.

The docs for `debugDisableShadows` should clearly articulate how it is expected to be used.

https://api.flutter.dev/flutter/painting/debugDisableShadows.html

### PR DESCRIPTION
**Pull request by Roshan13046**
## Description
Added explanation for debugDisableShadows property.

## Related Issues
Fixes  #69925

## Tests

I added the following tests: N/A because only docs are changed. 

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/painting/debug.dart [hunk 1]
```diff
@@ -13,6 +12,16 @@ import 'package:flutter/foundation.dart';
 /// This is useful when writing golden file tests (see [matchesGoldenFile]) since
 /// the rendering of shadows is not guaranteed to be pixel-for-pixel identical from
 /// version to version (or even from run to run).
+///
+/// It can be used within tests to enable shadows for a given test for screenshots.
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/debug.dart:16]: Isn't that pretty much exactly what the previous paragraph says?

File: packages/flutter/lib/src/painting/debug.dart [hunk 2]
```diff
@@ -13,6 +12,16 @@ import 'package:flutter/foundation.dart';
...
+///
+/// The only way this property works is if it is set to false in a single
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/debug.dart:18]: This is not true, there are other ways to use it. You could say something like: In those tests, this is usually set to false at the beginning of a test and back to true before the end of the test case. If it remains true when the test ends, an exception is thrown to avoid state leaking from one test case to another.

File: packages/flutter/lib/src/painting/debug.dart [hunk 3]
```diff
@@ -13,6 +12,16 @@ import 'package:flutter/foundation.dart';
...
+/// test case then set it back to true before the end of the test case.
+///
+/// If it is tried to alter outside of an individual test, an exception is thrown.
+///
+/// If it is not set back to true at the end of a test, an exception is thrown.
+
```
[the remark this answers is not in the document]
Comment (Roshan13046) [reply packages/flutter/lib/src/painting/debug.dart:24]: > nit: remove blank line

should I remove both blank lines ie lines no 22 and 24?
Thanks.

[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/src/painting/debug.dart:24]: Just the one in 24.

File: packages/flutter/lib/src/painting/debug.dart [hunk 4]
```diff
@@ -186,4 +192,4 @@ bool debugAssertAllPaintingVarsUnset(String reason, { bool debugDisableShadowsOv
     return true;
   }());
   return true;
-}
+}
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/debug.dart:195]: nit: can you revert this change?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
@Roshan13046 This is almost ready to go. Can you address the last comment?

**Comment by Roshan13046** [comment]
> @Roshan13046 This is almost ready to go. Can you address the last comment?

I have made the required changes.
Please check.
Thanks!

**Comment by goderbauer** [comment]
Tests seem to be failing now, probably because this branch is old. Can you rebase it to the latest master please? Sorry for the inconvenience...

**Comment by goderbauer** [comment]
@Roshan13046 looks like some tests are still failing due to an outdated master. Can you rebase this on top of the latest master? Thanks and sorry for the trouble.

**Comment by Roshan13046** [comment]
> @Roshan13046 looks like some tests are still failing due to an outdated master. Can you rebase this on top of the latest master? Thanks and sorry for the trouble.

I am a beginner in contributions.
Can you please tell me how to do that?

**Comment by goderbauer** [comment]
If you have your git setup as described in https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment you should be able to do `git pull upstream master`.
